test/utils/mdarray_Test.cpp

```diff
@@ -52,39 +62,44 @@ TEST(mdarray_Test, layout)
 {
     ngen::mdarray<int> a{{2, 2, 2, 2, 2}};
 
+    size_t indices0[] = {0, 0, 0, 0, 0};
+    size_t indices1[] = {1, 0, 0, 0, 0};
     ASSERT_EQ(
-        a.index({{0, 0, 0, 0, 0}}) + 1,
-        a.index({{1, 0, 0, 0, 0}})
+        a.index(indices0) + 1,
+        a.index(indices1)
     );
 
+    size_t indices0a[] = {0, 1, 1, 0, 1};
+    size_t indices1a[] = {1, 1, 1, 0, 1};
     ASSERT_EQ(
-        a.index({{0, 1, 1, 0, 1}}) + 1,
-        a.index({{1, 1, 1, 0, 1}})
+        a.index(indices0a) + 1,
+        a.index(indices1a)
     );
 
     ngen::mdarray<int> b{{2, 2, 2}};
 
+    size_t indices0b[] = {0, 0, 0};
+    size_t indices1b[] = {1, 0, 0};
     ASSERT_EQ(
-        b.index({{0, 0, 0}}) + 1,
-        b.index({{1, 0, 0}})
+        b.index(indices0b) + 1,
+        b.index(indices1b)
     );
 
+    size_t indices0c[] = {0, 1, 1};
+    size_t indices1c[] = {1, 1, 1};
     ASSERT_EQ(
-        b.index({{0, 1, 1}}) + 1,
-        b.index({{1, 1, 1}})
+        b.index(indices0c) + 1,
+        b.index(indices1c)
     );
 
     // add `i` to prevent braces around scalar warnings
     std::initializer_list<std::size_t> i = { 2 };
     ngen::mdarray<int> c{i};
 
+    size_t index0[] = {0};
+    size_t index1[] = {1};
     ASSERT_EQ(
-        c.index({{0}}) + 1,
-        c.index({{1}})
-    );
-
-    ASSERT_EQ(
-        c.index({{0}}) + 1,
-        c.index({{1}})
```

Comment:
This assertion was apparently duplicated? Maybe something different was intended when it was written @program-- ?

Reply:
I think the second assertion was supposed to be incremented one more time, since `c` is 1D of length 2. This entire test case is to ensure column-major form, so I might've just forgotten that individual case...